doc/index.rst

```diff
@@ -49,10 +49,14 @@ oriented interface or via a set of functions familiar to MATLAB users.
 Documentation
 -------------
 
-This is the documentation for Matplotlib version |version|.
+This is the documentation for Matplotlib version |version|. **For the
+documentation currently published by the ``master`` branch on Github,
```

Comment:
I do not think that documentation of an unreleased version should be linked prominently on the front page of the project. This is not relevant to the average user.

Reply:
It may well be, that opinions of devs are different. That's perfectly fine and is exactly why we have a review process so that we can agree on the best solution after discussion. I assume that the others will voice their opinion here as well. If you know who proposed this, you can ping them.

Reply:
https://26444-1385122-gh.circle-artifacts.com/0/home/circleci/project/doc/build/html/index.html   This doesn't build properly, but I think I agree w/ @timhoffm that this info should be on the Development page, not the front page.

Reply:
The reason I thought (still think) the dev documents should be linked prominently on the home page is because it's currently really hard for contributors to find (I've opened more than one issue against the docs only to learn that it's fixed on master). 
Long term there's likely a better solution, but for now I think a link underneath is fairly clear and unobtrusive. The [development page](https://matplotlib.org/devel/index.html) is a list of documents and I don't see a clear obvious place to put it unless we add a preface at the top of the page.

Reply:
The front page already has a link to the dev docs.  

I just pushed a commit to the repo that hosts the docs (https://github.com/matplotlib/matplotlib.github.com/commit/b28668599c6019632856400eff497ad47c252d53) to re-order and relabel that link.  While I agree this should be sign-posted, I agree with @timhoffm that we don't need it also bolded on the front page.